File: yh_tutorial/yh_tutorial_4/src/yh_pub_4.cpp
#include "ros/ros.h"
#include "yh_tutorial_4/yh_msg_4.h"

int main(int argc, char ** argv)
{
    ros::init(argc, argv, "yh_pub_4");
    ros::NodeHandle nh;
    
    ros::Publisher pub = nh.advertise<yh_tutorial_4::yh_msg_4>("yh_topic_4", 100);
    ros::Rate loop_rate(10);
    int cnt = 0;
    yh_tutorial_4::yh_msg_4 msg;

    while (ros::ok())
    {
        msg.stamp = ros::Time::now();
        msg.data = cnt;
        pub.publish(msg);
        cnt++;
        loop_rate.sleep();
    }

    return 0;
}

File: yh_tutorial/yh_tutorial_4/scripts/yh_pub_4.py
#!/usr/bin/python
# -*- coding: utf8 -*-

import rospy
from yh_tutorial_4.msg import yh_msg_4

def talker():
    pub = rospy.Publisher("yh_topic_4", yh_msg_4, queue_size=100)
    rospy.init_node("yh_pub_4", anonymous=True)
    loop_rate = rospy.Rate(10)
    cnt = 0
    msg = yh_msg_4()

    while not rospy.is_shutdown():
        msg.stamp = rospy.get_rostime()
        msg.data = cnt
        pub.publish(msg)
        cnt += 1
        loop_rate.sleep()

if __name__ == "__main__":
    try:
        talker()
    except rospy.ROSInterruptException:
        pass